downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/GOAGeneratorUtilities.java

```diff
@@ -0,0 +1,247 @@
+            instanceDate = getDate(createdInst);
+        }
+
+        // Stores date in global hash that allows date value to be updated if a more recent date was found.
+        if (dates.get(goaLine) == null || instanceDate > dates.get(goaLine)) {
+            dates.put(goaLine, instanceDate);
+        }
+        return instanceDate;
+    }
+
+    /**
+     * Retrieves date from instance and formats it for GO annotation file.
+     * @param instanceEditInst -- GKInstance, instanceEdit from either a Modified or Created instance.
+     * @return -- Integer, from instanceEdit's dateTime. Parsed to remove the Timestamp.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    private static int getDate(GKInstance instanceEditInst) throws Exception {
+        return Integer.valueOf(instanceEditInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
+    }
+
+    /**
+     * Iterates through the lines in the 'goaLines' list, retrieves the date associated with that line and also adds the 'Reactome' column before adding it to the gene_association.reactome file.
+     * @throws IOException -- File writing/reading exceptions.
+     */
+    public static void outputGOAFile() throws IOException {
+
+        final String GAF_HEADER = "!gaf-version: 2.1\n";
+
+        Files.deleteIfExists(Paths.get(GOA_FILENAME));
+        Files.write(Paths.get(GOA_FILENAME), GAF_HEADER.getBytes());
+
+        List<String> lines = goaLines.stream().sorted().map(
+                line -> String.join("\t", line, dates.get(line).toString(), REACTOME_STRING, "","")
+        ).collect(Collectors.toList());
+        Files.write(Paths.get(GOA_FILENAME), lines, StandardOpenOption.APPEND);
+
+        gzipGOAFile();
+    }
+
+    /**
+     * Gzips gene_association.reactome file
+     * @throws IOException -- File writing/reading exceptions.
+     */
+    private static void gzipGOAFile() throws IOException {
+        File goaFile = new File(GOA_FILENAME);
+        File goaFileGZipped = new File(GOA_FILENAME + ".gz");
+        GZIPOutputStream gzipOutputStream = new GZIPOutputStream(new FileOutputStream(goaFileGZipped));
+        try (FileInputStream fileInputStream = new FileInputStream(goaFile)) {
+            byte[] buffer = new byte[1024];
+            int length;
+            while((length=fileInputStream.read(buffer)) != -1) {
+                gzipOutputStream.write(buffer, 0, length);
+            }
+            gzipOutputStream.close();
+        }
+    }
+
+    /**
+     * Move file into DownloadDirectory folder corresponding to release number.
+     * @param targetDirectory -- String, where the file will be moved to.
+     * @throws IOException -- If file or targetDirectory do not exist, this will be thrown.
+     */
+    public static void moveFile(String targetDirectory) throws IOException {
+        String updatedTargetDirectory = targetDirectory + GOA_FILENAME + ".gz";
```

Comment:
Could this be called 'targetFilePath' since it is the directory plus the file name?